### ISSUE TITLE
Feature request: Allow formatting unsaved files

### ISSUE THREAD

**Issue by RandomSpaceship** [isbecker/treefmt-vscode#4]
Currently the extension can only format saved files, which feels highly unintuitive - it leads to an experience where you can hit format document and nothing happens (in the event that you've made changes which require a re-format).

**Comment by isbecker** [comment]
@RandomSpaceship, thank you for the request.

In my experience, formatting tools (`go fmt`, `ruff fmt`, `prettier`, etc) - which are the targets for `treefmt` - all require the file to be saved prior to running. If you're looking for a "current buffer" formatter, I think that your use-case is not covered under the umbrella of this extension (or treefmt). I understand your use-case, and I agree that the experience with vscode can be a tad buggy. It is a tradeoff that is the nature of using these formatter tools inside vscode.

Or, I might be misunderstanding. There may be a way to configure everything to work the way you want. It will be tool dependent, whether or not they support passing data (I guess via stdin), or we could pass a /tmp file to treefmt to get a result and then write that back to the buffer. Just some ideas that may be possible to make what you want happen.

But, for now, I think those are beyond the scope of complexity that I want to have in the extension.

If you disagree with my assessment, please let me know, but I don't plan to work on it right now. I'm open to PRs though!

**Comment by RandomSpaceship** [comment]
Fair enough. Supposedly treefmt [supports](https://treefmt.com/latest/getting-started/usage/#format-stdin) formatting from stdin, but I have no idea if that works for all formatters, or just those with stdin support natively - I haven't tested it yet.

It's honestly a pretty minor thing overall, and having to save a bit more often really isn't the end of the world. If I have some spare time and feel like learning about vscode extensions I'll probably raise a PR, but don't hold high hopes.

**Comment by isbecker** [comment]
@RandomSpaceship I couldn't help myself. Once you mentioned that there is a `--stdin` option for treefmt, I tried it out and it worked without needing to modify my treefmt.toml to accommodate, so I figured it would be a simple task to add a new command.

I made a PR #6 

Please give it a look and you can try it out locally yourself, if you like. For now, it has only a command interface for calling the feature. Meaning, you can do a `Ctrl+Shift+P` (or Command on mac), and select `Run Treefmt with Stdin` to try the new feature.

**Comment by RandomSpaceship** [comment]
Just tested it, and it seems to work really well. Funny enough, I think it's actually faster in my use case because the C/C++ extension sometimes takes a while to finish processing, blocking VSCode from saving (very large include path causes extremely slow workspace processing because it all runs on a single thread).

Formatters tested, in no particular order:
- `ruff format`
- `taplo`
- `typos`
- `yamlfmt`
- `prettier`
- `nixfmt` (rfc style)
- `shellfmt`
- `prettier`
- And probably a couple others I forgot about
All seems to work exactly as expected.

**Comment by isbecker** [comment]
Thanks for testing all those formatters

I think as I mentioned in the PR, I will likely make the no-save the default behavior. 

I'm considering keeping the on-save based code available through a command or maybe a boolean flag, just in case anyone needs it. I'm hesitant about removing functionality entirely without warning.
